### PR TITLE
Making info properties clear

### LINKED ISSUE #2144: make worksheet_info properties clearer
the names are confusing because "raw" refers to different things in different properties:
raw: refers to source lines in editor
raw_to_block, block_to_raw: "raw" refers to raw items (search directive lines have been expanded already)
Also the documentation for these mappings are lacking, which causes great confusion for people working on them

### PR DESCRIPTION
**Pull request by yipenghe**
Fix #2144 

The main goal of this PR is make several properties of worksheet_info more clearer to what they mean
- raw => source : this makes it consistent to what we call it in the frontend as in "edit source"

- raw_to_block, block_to_raw : since we change raw=>source, raw in these variables here will solely mean raw items that were interpreted from source lines

- documentation: I added quite a bit of documentation to what raw_to_block, block_to_raw mean, mainly because when I was working on the frontend, these mappings were very confusing, and were used to address several focus setting issues

- add raw_items_to_source_index: this is used for copy/paste/cut and will be used be schema items, adding it in this PR so later PRs can all use it 

- items: items should really be named "block_items" to avoid confusion, the only thing is I need to make sure ["items"] in other places of the codebase doesn't get affected

### CODE REVIEW

File: codalab/rest/interpret.py
Comment (yipenghe) [inline codalab/rest/interpret.py]: I removed this function because it doesn't really provide much help in understanding (the line is simple enough to know its expanding all raw_items), moreover we need to get the item_to_source mapping which requires two return values if we keep this function

File: codalab/lib/spec_util.py
Comment (yipenghe) [inline codalab/lib/spec_util.py]: black command did this

File: codalab/rest/interpret.py
Comment (yipenghe) [inline codalab/rest/interpret.py]: will remove

File: codalab/rest/interpret.py
Comment (yipenghe) [inline codalab/rest/interpret.py:161]: Looking for suggestions on the documentation, the worksheet_info contains a lot of complexity so we really need a accurate and helpful documentation for later development

**Comment by yipenghe** [comment]
This is kind of an important change (in my opinion) and I hope to make sure the change will actually make things clearer to what all the maps in worksheet_info mean and used for

File: codalab/rest/interpret.py
Comment (epicfaace) [inline codalab/rest/interpret.py]: If we've renamed "raw" to "source", then we might want to replace "raw_to_block" to "source_to_block", etc.

File: codalab/rest/interpret.py
Comment (percyliang) [inline codalab/rest/interpret.py]: why `-list`?

File: codalab/rest/interpret.py
Comment (percyliang) [reply codalab/rest/interpret.py:161]: Totally agree - we should probably make a Python object out of this (e.g., using `dataclass`)

File: codalab/rest/interpret.py
Comment (yipenghe) [reply codalab/rest/interpret.py]: raw_to_block is actually raw items to block, not source

File: codalab/rest/interpret.py
Comment (percyliang) [reply codalab/rest/interpret.py:161]: Probably should give broader context (either here or refer people to another place).
- source (raw worksheet_items in the DB, rendered as lines) [give example]
- blocks (semantic units that are rendered).

I wouldn't call it "block items" - I think items should always refer to raw items (because that's in the DB, which is annoying to change); just call them blocks everywhere.

**Review by percyliang — CHANGES_REQUESTED** [review]
Definitely we should clean this up, thanks for doing this!  Some comments on how we can get to more consistent naming.

File: codalab/rest/interpret.py
Comment (yipenghe) [reply codalab/rest/interpret.py]: by raw items, it means items rendered by the source but not grouped into block yet

File: codalab/rest/interpret.py
Comment (yipenghe) [inline codalab/rest/interpret.py:282]: After this line, the original worksheet_info['items'] are actually blocks

File: frontend/src/components/worksheets/NewRun/NewRun.jsx
Comment (yipenghe) [inline frontend/src/components/worksheets/NewRun/NewRun.jsx:284]: frontend code: just changing items=> blocks

File: codalab/rest/interpret.py
Comment (teetone) [inline codalab/rest/interpret.py]: Should it be "has 2 rows"?

**Review by teetone — APPROVED** [review]

File: codalab/rest/interpret.py
Comment (armantajback) [inline codalab/rest/interpret.py]: nit: revise casing through this block comment

**Comment by yipenghe** [comment]
@percyliang  The merge is blocked by the outdated "request changes"

**Comment by yipenghe** [comment]
@percyliang Can you re-review this / dismiss the request change?

**Review by percyliang — APPROVED** [review]
Great!

**Pull request merged**
